src/backend/tcop/utility.c

```diff
@@ -362,19 +362,64 @@ ProcessUtility(Node *parsetree,
 {
 	Assert(queryString != NULL);	/* required as of 8.4 */
 
-	/*
-	 * We provide a function hook variable that lets loadable plugins get
-	 * control when ProcessUtility is called.  Such a plugin would normally
-	 * call standard_ProcessUtility().
-	 */
-	if (ProcessUtility_hook)
-		(*ProcessUtility_hook) (parsetree, queryString,
-								context, params,
-								dest, completionTag);
-	else
-		standard_ProcessUtility(parsetree, queryString,
-								context, params,
-								dest, completionTag);
+	int saved_command_id = MyProc->queryCommandId;
+
+	if (Gp_role != GP_ROLE_EXECUTE)
+		increment_command_count();
+
+#ifdef FAULT_INJECTOR
+	if (SIMPLE_FAULT_INJECTOR("track_query_command_id") == FaultInjectorTypeSkip ||
+		SIMPLE_FAULT_INJECTOR("track_query_command_id_at_start") == FaultInjectorTypeSkip)
+		elog(NOTICE, "START %s | Q: %s | QUERY ID: %d",
+										__FUNCTION__,
+										queryString,
+										MyProc->queryCommandId);
+#endif
```

Comment:
Why not make the functions and macros from the executor public and use them here too?

Reply:
Functions and macros from the executor receives `QueryDesc` as an argument. But we do not have it in the `ProcessUtility()`. Of course it is always possible to rework the functions to get rid of the `QueryDesc` and just pass more params, but it will make these utility functions more bulky from my point of view, and I preferred to keep them simpler and write specific code in  `ProcessUtility()`.